Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1

```diff
@@ -0,0 +1,111 @@
+BeforeAll {
+    Import-Module (Join-Path -Path $PSScriptRoot -ChildPath '../../../../PowerShell/ScubaGear/Modules/CreateReport')
+    New-Item -Path (Join-Path -Path $PSScriptRoot -ChildPath "./CreateReportStubs") -Name "CreateReportUnitFolder" -ErrorAction SilentlyContinue -ItemType Directory | Out-Null
+    New-Item -Path (Join-Path -Path $PSScriptRoot -ChildPath "./CreateReportStubs/CreateReportUnitFolder") -Name "IndividualReports" -ErrorAction SilentlyContinue -ItemType Directory | Out-Null
+}
+
+Describe -Tag CreateReport -Name 'New-Report' {
+    Context "Light mode case" {
+        BeforeAll {
+            [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'ProductNames')]
+            $ProductNames = @("teams", "exo", "defender", "aad", "powerplatform", "sharepoint", "onedrive")
+            [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'ArgToProd')]
+            $ArgToProd = @{
+                teams         = "Teams";
+                exo           = "EXO";
+                defender      = "Defender";
+                aad           = "AAD";
+                powerplatform = "PowerPlatform";
+                sharepoint    = "SharePoint";
+                onedrive      = "OneDrive";
+            }
+            [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'ProdToFullName')]
+            $ProdToFullName = @{
+                Teams         = "Microsoft Teams";
+                EXO           = "Exchange Online";
+                Defender      = "Microsoft 365 Defender";
+                AAD           = "Azure Active Directory";
+                PowerPlatform = "Microsoft Power Platform";
+                SharePoint    = "SharePoint Online";
+                OneDrive      = "OneDrive for Business";
+            }
+            $IndividualReportPath = (Join-Path -Path $PSScriptRoot -ChildPath "./CreateReportStubs/CreateReportUnitFolder/IndividualReports")
+            [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'CreateReportParams')]
+            $CreateReportParams = @{
+                'IndividualReportPath' = $IndividualReportPath;
+                'OutPath'              = (Join-Path -Path $PSScriptRoot -ChildPath "./CreateReportStubs");
+                'OutProviderFileName'  = "ProviderSettingsExport";
+                'OutRegoFileName'      = "TestResults";
+                'DarkMode'             = $false;
+            }
+        }
+        It 'Creates a report for Azure Active Directory' {
```

Comment:
I would expect additional tests - for example, since this is AAD the warnings are treated different from other products.  We should check that logic is being executed properly.

Reply:
Thanks for pointing this out. Unit tests are a moving target and expect to keep working and integrating more error handling and QOL improvements